From f9ab0e3e7d96f5375d493e395901513fba746b23 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 7 Nov 2016 16:41:01 +0000
Subject: [PATCH 25/35] GWT: Debug rpc issues

For some reason my RPC uri is coming through to SessionMain.cpp as
`//rpc/some_function` instead of `/rpc/some_function`. This should
help debug that.
---
 src/cpp/desktop/DesktopBrowserWindow.cpp                              | 2 ++
 src/cpp/desktop/DesktopSessionLauncher.cpp                            | 2 ++
 src/cpp/desktop/DesktopWebPage.cpp                                    | 4 ++++
 src/cpp/session/SessionMain.cpp                                       | 2 ++
 src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java           | 4 +++-
 src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java | 1 +
 6 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/src/cpp/desktop/DesktopBrowserWindow.cpp b/src/cpp/desktop/DesktopBrowserWindow.cpp
index 6ed75e3863..9a22d20e40 100644
--- a/src/cpp/desktop/DesktopBrowserWindow.cpp
+++ b/src/cpp/desktop/DesktopBrowserWindow.cpp
@@ -37,6 +37,8 @@ BrowserWindow::BrowserWindow(bool showToolbar,
    adjustTitle_ = adjustTitle;
    progress_ = 0;
 
+   qDebug() << "BrowserWindow::BrowserWindow: name:" << name << " baseUrl: " << baseUrl.toString() << " allowExternalNavigate: " << allowExternalNavigate;
+
    pView_ = new WebView(baseUrl, this, allowExternalNavigate);
    connect(pView_, SIGNAL(titleChanged(QString)), SLOT(adjustTitle()));
    connect(pView_, SIGNAL(loadProgress(int)), SLOT(setProgress(int)));
diff --git a/src/cpp/desktop/DesktopSessionLauncher.cpp b/src/cpp/desktop/DesktopSessionLauncher.cpp
index 6e32229408..426dc4f15e 100644
--- a/src/cpp/desktop/DesktopSessionLauncher.cpp
+++ b/src/cpp/desktop/DesktopSessionLauncher.cpp
@@ -420,6 +420,8 @@ void SessionLauncher::buildLaunchContext(QString* pHost,
    *pUrl = QUrl(QString::fromUtf8("http://") + *pHost +
                 QString::fromUtf8(":") + *pPort + QString::fromUtf8("/"));
 
+   qDebug() << "SessionLauncher::buildLaunchContext: pUrl->toString(): " << pUrl->toString() << " pHost: " << *pHost << " pPort: " << *pPort;
+
    if (!confPath_.empty())
    {
       *pArgList << QString::fromUtf8("--config-file") <<
diff --git a/src/cpp/desktop/DesktopWebPage.cpp b/src/cpp/desktop/DesktopWebPage.cpp
index cf76d6884b..6f248da51d 100644
--- a/src/cpp/desktop/DesktopWebPage.cpp
+++ b/src/cpp/desktop/DesktopWebPage.cpp
@@ -220,6 +220,8 @@ void WebPage::javaScriptConsoleMessage(JavaScriptConsoleMessageLevel /*level*/,
 
 bool WebPage::acceptNavigationRequest(const QUrl &url, NavigationType navType, bool isMainFrame)
 {
+   qDebug() << "WebPage::acceptNavigationRequest: " << url.toString();
+
    if (url.toString() == QString::fromUtf8("about:blank"))
       return true;
 
@@ -389,6 +391,8 @@ void WebPage::handleBase64Download(QUrl url)
 
 void WebPage::setViewerUrl(const QString& viewerUrl)
 {
+   qDebug() << "WebPage::setViewerUrl: " << viewerUrl;
+
    // record about:blank literally
    if (viewerUrl == QString::fromUtf8("about:blank"))
    {
diff --git a/src/cpp/session/SessionMain.cpp b/src/cpp/session/SessionMain.cpp
index c2322bd2ca..c82dbe4157 100644
--- a/src/cpp/session/SessionMain.cpp
+++ b/src/cpp/session/SessionMain.cpp
@@ -28,6 +28,8 @@
 #include <csignal>
 #include <limits>
 
+#include <iostream>
+
 #include <boost/shared_ptr.hpp>
 #include <boost/function.hpp>
 #include <boost/lexical_cast.hpp>
diff --git a/src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java b/src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java
index 6ff19caad0..5e1db4eda5 100644
--- a/src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java
+++ b/src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java
@@ -29,7 +29,7 @@ import org.rstudio.core.client.jsonrpc.RequestLogEntry.ResponseType;
 // for retries after network or authentication errors)
 public class RpcRequest 
 {
-   public static final boolean TRACE = false ;
+   public static final boolean TRACE = true ;
    
    public RpcRequest(String url, 
                      String method, 
@@ -92,6 +92,8 @@ public class RpcRequest
       try
       {
          String requestString = request.toString();
+         if (TRACE)
+            Debug.log("RpcRequest url_: " + url_) ;
          if (TRACE)
             Debug.log("Request: " + requestString) ;
 
diff --git a/src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java b/src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java
index 36351ec2d0..ad05cd3741 100644
--- a/src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java
+++ b/src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java
@@ -433,6 +433,7 @@ public class RemoteServer implements Server
       ensureListeningForEvents();
       
       // return the url
+      Debug.log("GWT.getHostPageBaseURL(): " + GWT.getHostPageBaseURL()) ;
       return GWT.getHostPageBaseURL() + pathName;
    }
    
-- 
2.14.3 (Apple Git-98)

